### DISCUSSION TITLE
indexing certain orientation

### DISCUSSION THREAD

**Discussion by sarialzerey** [mtex-toolbox/mtex#1918] [Ask Anything] [answered]
Hi,

It might be easy for others but I have been struggling with this for over 2 weeks.
Can anyone help me please to do the following functions:
1- Highlighting (111) orientation grains with 10 degrees tolerance in a 'copper' phase and showing the rest of the grains in gray color.
2- Calculating the area fraction of (111) among all copper orientations.
3- Calculating the area of (111) orientations.

Thanks in advance
Best,
Sari

**Comment by ralfHielscher** [comment]
Hi Sari,

I think you need to explain what an (111) orientation is. Do you mean an orientation with (111) being parallel to the z-axis?

Ralf.

**Comment by sarialzerey** [comment]
Hi Ralf,

thank you for responding. I mean the out of plan direction of the crystal is parallel to Z direction. In other words, (111) plane for that orientation is parallel to XY plane. I would highly appreciate your support. 

Best,
Sari

**Comment by ralfHielscher** [comment]
1.
```matlab
grains = grains('copper')
csCo = grains.CS;
oriCo = grains('copper').meanOrientation;
ind = angle(inv(oriCo * zvector, Miller(1,1,1,csCo)) < 10*degree;
plot(grains(ind),grains(ind).meanOrientation)
plot(grains(~ind),'faceColor','gray')
```
2.
```matlab
grains(ind).area / grains.area
```
3.
```matlab
grains(ind).area
```
Ralf.

**Comment by sarialzerey** [comment]
I keep receiving error message: 

Warning: Possibly applying an orientation to an object in specimen coordinates! 
Error using inv
Too many input arguments.

Error in untitled4 (line 5)
ind = angle(inv(oriCo * zvector, Miller(1,1,1,csCo))) < 15*degree;

**Reply by ralfHielscher** [reply, marked as answer]
Sorry, this was a typo in my answer, it should be
```matlab
ind = angle(inv(oriCo) * zvector, Miller(1,1,1,csCo)) < 15*degree;
```
Ralf.

**Comment by sarialzerey** [comment]
Hi Ralf,

Thank you so much for the support. Your code worked. I really appreciate it.

Best,
Sari
